src/app/static/src/app/components/downloadResults/UploadModal.vue

```diff
@@ -0,0 +1,112 @@
+                    <span>{{ dataset }}</span></div>
+                <div id="instructions" class="mt-3" v-translate="'uploadFileInstruction'"></div>
+                <div id="output-file-id" class="mt-3" v-for="uploadFile in uploadFiles" :key="uploadFile.index">
+                    <div class="mt-3 form-check">
+                        <input class="form-check-input"
+                               type="checkbox"
+                               :value="uploadFile.index"
+                               v-model="uploadFilesToAdr"
+                               :id="`id-${uploadFile.index}`">
+
+                        <label class="form-check-label"
+                               :for="`id-${uploadFile.index}`"
+                               v-translate="uploadFile.displayName"></label>
+
+                        <small v-if="uploadFile.resourceId" class="text-danger row">
+                        <span class="col-auto">
+                        <span v-translate="'uploadFileOverwrite'"></span>{{ lastModified(uploadFile.lastModified) }}
+                        </span>
+                        </small>
+                    </div>
+                </div>
+                <div class="mt-3">
+                    <label for="description-id" v-translate="'uploadFileDesc'"></label>
+                    <textarea v-model="uploadDescToAdr"
+                              class="form-control" rows="3"
+                              id="description-id"></textarea>
+                </div>
+            </div>
+            <template v-slot:footer>
+                <button
+                    type="button"
+                    class="btn btn-red"
+                    v-translate="'ok'"></button>
+                <button
+                    type="button"
+                    class="btn btn-white"
+                    @click.prevent="handleCancel"
+                    v-translate="'cancel'"></button>
+            </template>
+        </modal>
+    </div>
+</template>
+
+<script lang="ts">
+    import Vue from "vue";
+    import Modal from "../Modal.vue";
+    import {Dict, UploadFile} from "../../types";
+    import {BaselineState} from "../../store/baseline/baseline";
+    import {formatDateTime, mapStateProp, mapStateProps} from "../../utils";
+    import {ADRState} from "../../store/adr/adr";
+
+    interface Methods {
+        handleCancel: () => void
+        lastModified: (date: string) => string | null
+    }
+
+    interface Computed {
+        dataset: string
+        uploadFiles: Dict<UploadFile>
+    }
+
+    interface Data {
+        uploadFilesToAdr: []
+        uploadDescToAdr: string
+    }
+
+    interface Props {
+        open: boolean
+    }
+
+    export default Vue.extend<Data, Methods, Computed, Props>({
+        name: "UploadModal",
+        props: {
+            open: {
+                type: Boolean
+            }
+        },
+        data(): Data {
+            return {
+                uploadFilesToAdr: [],
```

Comment:
As mentioned above, I think this should be:
```suggestion
                uploadFilesToAdr: string[],
```